### PR TITLE
CMake integration

### PR DESCRIPTION
**Pull request by eidelen**
Hi wichtounet,
I added a cmake-config with a little example to your mnist project. Like that, I can easy use mnist without copying it somewhere to my source tree. Also the data's directory is passed by cmake.
Would be nice if you can take it in. However, feel free to reject :)
Gruss Adrian

PS: Thanks for your work. I could profit a lot.

### CODE REVIEW

**Comment by wichtounet** [comment]
Hi, 

Thanks for the contribution. It's very interesting. 

Two things before I merge.

1. Could you use MIT License for the .cpp example ? I'd rather not mix licenses in such a small project.  
2. Very small: could you rename mnistexample into mnist_example ? 

Thanks

**Comment by eidelen** [comment]
Nice :) I updated the license and the example name.
Have a good time,
Adrian

File: example/main.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+#include <iostream>
+#include "mnist/mnist_reader.hpp"
+
+int main(int argc, char* argv[]) {
+    // MNIST_DATA_LOCATION set by MNIST cmake config
+    std::cout << "MNIST data directory: " << MNIST_DATA_LOCATION << std::endl;
+
+    // Load MNIST data
+    mnist::MNIST_dataset<std::vector, std::vector<uint8_t>, uint8_t> dataSet =
```
Comment (wichtounet) [inline example/main.cpp]: Just to be coherent with my naming convention, could you rename to dataset ?

**Comment by wichtounet** [comment]
I just missed that one before. No big deal again :P

Thanks for the changes. I'll merge as soon as this last point is fixed.

**Comment by eidelen** [comment]
Thanks :)

**Pull request merged**